### DISCUSSION TITLE
Does Mirth Connect support xDT?

### DISCUSSION THREAD

**Discussion by tGitm** [nextgenhealthcare/connect#4803] [Q&A] [answered]
Could you tell me if Mirth Connect supports xDT (GDT, LDT, BDT) and a conversion from xDT (GDT, LDT, BDT) to XML and vice versa as well?

**Comment by tonygermano** [comment, marked as answer]
That's not a recognized data type. Do you have a specification for the format?

**Reply by tGitm** [reply]
https://forums.mirthproject.io/forum/mirth-connect/support/9954-does-mirth-connect-support-xdt

There was the same answer, and someone mentioned something about delimited text data type.

**Reply by tonygermano** [reply]
I've never heard of that format before, so I can't tell you if the delimited text data type will work or not if you don't have more information about the format.

**Reply by tGitm** [reply]
It's a German healthcare data format, like HL7, here's a example of one:

01380006302 014810000534 0278315Empfaenger-Kennung 0178316_TOMEDO_ 01092063 01292182.1 011300015 0153000URO-15 0133101Ivan 0153102Testni 017310309081980 0153103090880 017310309.08.80 017310309081980 019310309.08.1980 017310319800809 01031101 0103110M 024310632479 Bollacken 0203107dsdsadsa 33 0183113Bollacken 014311232479 017310512343243 0148402EKG00 0183626432432432 0183629432432423 0103752p 0243633ivan@testni.com 014299910000 0122998ECG 026feldkennungfeldinhalt 0262002Unknown insurance 0108609P

**Reply by tonygermano** [reply]
Do you have a formal specification or implementation guide? Hoping for something in English is probably too much? I can't make any sense of what I'm looking at. I'm guessing all of those numbers have some meaning. It is really supposed to all be on one line?

**Reply by tGitm** [reply]
Sorry but that's all I got from my company. No it's not 1 line, every data in it's row, like that:
![image](https://user-images.githubusercontent.com/47832737/138478785-8bbfa782-b9aa-474f-be55-25965815acac.png)

**Reply by jonbartels** [reply]
https://github.com/albertzak/xdt/tree/master/test/fixtures is a parser project and has a few samples.

@tGitm - First a disclaimer - I know absolutely nothing about this format. Think and verify any advice I give as it may be wrong.

If you look at the source code (and if you trust this open source parser!) you can see that it's sort of acting like a delimited format - https://github.com/albertzak/xdt/blob/master/main.js . Node.js and JS scripts inside MC are not the same. They are somewhat similar and you might be able to borrow some ideas from it.

It looks like a numeric-index. I'm not quite sure how the length factors in. 

To read this format one definitely has to have a specification for what the field number represents. 

Even though MC doesn't support this format directly, it should be possible to set up MC to parse these messages with transformer scripts and generate XML output.

**Reply by tGitm** [reply]
Thanks, that's what I'll try! I think that this can be marked as answer.

**Reply by jonbartels** [reply]
https://ec.europa.eu/eip/ageing/standards/ict-and-communication/other-ict/xdt_en.html (emphasis mine)

> A message consists of several “records”, while each record consists of one or more “fields”, each field being encoded as one line of text. Each line starts with a three-digit number indicating the length of the line (including the carriage return/line feed), immediately followed by four digits indicating the field ID, followed by the value. **The standard precisely defines the permitted sequence of fields for each record type, including optionality, possible repetitions, sub-fields etc**

So it seems easy to parse each message and line, the meaning comes from the standard. Most of the links I'm finding to the standard do not work. I could probably scrape by even with a German language copy

**Reply by tGitm** [reply]
Ou fine, sounds easy to parse, thanks again!

**Comment by jonbartels** [comment]
Is xDT a datatype or is it also a transmission protocol? 

Do you have any links or references that an MC expert who doesn't know about xDT could read to learn about the format?

**Reply by tGitm** [reply]
It's a German healthcare data format, like HL7, here's a example of one:

``
01380006302
014810000534
0278315Empfaenger-Kennung
0178316_TOMEDO_
01092063
01292182.1
011300015
0153000URO-15
0133101Ivan
0153102Testni
017310309081980
0153103090880
017310309.08.80
017310309081980
019310309.08.1980
017310319800809
01031101
0103110M
024310632479 Bollacken
0203107dsdsadsa 33
0183113Bollacken
014311232479
017310512343243
0148402EKG00
0183626432432432
0183629432432423
0103752p
0243633ivan@testni.com
014299910000
0122998ECG
026feldkennungfeldinhalt
0262002Unknown insurance
0108609P
``

**Reply by renemewes** [reply]
@tGitm  -  Also, please take a look at this commercial plugins: https://www.vireq.ch/product-page/mirth-connect-ce-gdt & https://www.vireq.ch/product-page/mirth-connect-ce-ldtx

**Reply by tGitm** [reply]
Thanks for this! I'll check it out.
